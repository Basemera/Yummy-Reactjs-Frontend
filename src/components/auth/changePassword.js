import React, { Component } from 'react';

class ChangePassword extends Component {
  render() {
    return (
      <div className="background">
        <div id="changepassword">
          <div className="form-group">
            <h1> You Wanna change your password, Go ahead!</h1>
            <form onSubmit={this.handleLogin} >
                  <div className="input-group">
          <span className="input-group-addon" id="email"><i className="fa fa-envelope" /></span>
          <input
                  type="email"
                  id="emailsignup"
                  name="email_field"
                  aria-describedby="email"
                  className="form-control"
                  placeholder="example@gmail.com"
                />
        </div>
                  <br />
                  <div className="input-group">
          <span className="input-group-addon" ><i className="fa fa-lock" /></span>
          <input
                  type="password"
                  name="password_field"
                  className="form-control"
                  placeholder="Password"
                />
        </div>
                  <br />
                  <div className="input-group">
          <span className="input-group-addon" ><i className="fa fa-lock" /></span>
          <input
                  type="password"
                  name="new_password"
                  className="form-control"
                  placeholder="New Password"
                />
        </div>
                  <br />
                  <input name="signin" className="btn-nav" id="signin" value="Save" type="submit" />
                </form>
          </div>
        </div>
      </div>
    );
  }
}
export default ChangePassword;
