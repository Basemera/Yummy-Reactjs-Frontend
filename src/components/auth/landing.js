import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import Slideshow from '../slideshow';


class Landing extends React.Component {
  static propTypes = {
    loggedIn:PropTypes.bool,
  }

  static defaultProps = {
    loggedIn:false,
  }
  componentDidMount() {
    if (this.props.loggedIn) {
      this.props.history.push('/dashboard');
    }
  }
  render() {
    return (
      <Slideshow />
    );
  }
}

export class Clock extends React.Component {
  constructor(props) {
    super(props);
    const currentTime = new Date();
    this.state = {
      hours: currentTime.getHours(),
      minutes: currentTime.getMinutes(),
      seconds: currentTime.getSeconds(),
      ampm: currentTime.getHours() >= 12
        ? 'pm'
        : 'am',
    };
    this.setTimer();
  }

  setTimer() {
    setTimeout(this.updateClock.bind(this), 1000);
  }
  componentWillUnmount() {
    if (this.timeout) {
      clearTimeout(this.timeout);
    }
  }
  updateClock() {
    const currentTime = new Date();
    this.setState({
      hours: currentTime.getHours(),
      minutes: currentTime.getMinutes(),
      seconds: currentTime.getSeconds(),
      ampm: currentTime.getHours() >= 12
        ? 'pm'
        : 'am',
    }, this.setTimer);
  }

  render() {
    const {
      hours, minutes, seconds, ampm,
    } = this.state;
    return (
      <div className="clock">
        {hours === 0
          ? 12
          : (hours > 12)
            ? hours - 12
            : hours
}:{minutes > 9
          ? minutes
          : `0${minutes}`
}:{seconds > 9
          ? seconds
          : `0${seconds}`
} {ampm}
      </div>
    );
  }
}

export const Nav = props => (
  <div id="header">
    <div className="container">
      <div className="logo" />
      <div id="author">
        <h3>Recipes by KH</h3>
      </div>
      <div id="menu">
        {!props.loggedIn
          ? (
            <Fragment>
              <Link to="/login" id="signin" className="btn btn-nav">
              <i className="fa fa-sign-in "/>Login</Link>
              <Link to="/register" id="signup" className="btn btn-nav">
              <i className="fa fa-user-plus "/>
                Signup
              </Link>
              <Link to="/" id="homebtn" className="btn btn-nav">
              <i className="fa fa-home "/>Home</Link>
            </Fragment>
          )
          : (
            <div>
              <Link  to="/dashboard" className="btn btn-primary">
                  Categories
                  <i className="glyphicon glyphicon-cutlery"/>
              </Link>
              <Link className="btn btn-success" to="/about">About Us <i className="glyphicon glyphicon-tags"/></Link>
              
              <Link className="btn btn-danger" to="/" onClick={(e) => { props.logout(e); }}>
              Logout <i className="fa fa-sign-out"/></Link>
            </div>

          )}
      </div>
    </div>
  </div>

);

Nav.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  logout: PropTypes.func.isRequired,
};

export default Landing;
