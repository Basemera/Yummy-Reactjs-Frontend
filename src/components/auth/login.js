
import React, { Component } from 'react';
import axios from 'axios';
import {notify} from 'react-notify-toast'
import { Link } from 'react-router-dom';

class Login extends Component {
  state = {
    password_field: '',
    email_field: '',
}

handleInputChange = (event)=>{
    const { name, value} = event.target;
    this.setState({
        [name]:value,
    });
}

handleLogin = (event)=>{
    const {email_field:email, password_field:password} = this.state;

    event.preventDefault();
    axios.post('http://127.0.0.1:5000/api/v1/auth/login',{email,password})
    .then(response=>{
        notify.show(response.data.message, 'success', 4000);
        this.props.history.push('/dashboard');
        localStorage.setItem('accessToken', response.data.access_token);
        console.log(response);
    }).catch(error=>{
        if(error.response){
            alert(error.response.data.message)
        }else if (error.request){
            alert("Request not made")
        }
    });
}
  render() {
    const {password_field, email_field} = this.state;
    return (
      <div className="background">
        <div id="logbar">
        <div className="form-group">
          <h1> LOGIN HERE</h1>
          <form onSubmit={this.handleLogin} >
            <label>Email</label>
            <input type="email" name="email_field" className="form-control" onChange={this.handleInputChange} value={email_field}/>
            <label>Pass word</label>
            <input type="password" name="password_field"  className="form-control" onChange={this.handleInputChange} value={password_field}/>
            <br/>
            <input name="signin" id="signin" value="LOGIN" type="submit"/>
          </form>
          <div className="change_link">
				Not a member yet ?
				<Link to="/register">< h4 className="toregister" >Join us</h4></Link>
								</div>
        </div>
        </div>
        <footer className="footer">
          <p>&copy; Hadijah 2018</p>
        </footer>
      </div>
    );
  }
}

export default Login;
