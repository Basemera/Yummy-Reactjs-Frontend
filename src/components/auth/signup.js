import React, {Component} from 'react';
import axiosInstance from '../Apicalls';
import {notify} from 'react-notify-toast'
import {Link} from 'react-router-dom';

class Signup extends Component {
    state = {
        password_field: '',
        email_field: ''
    }

    handleInputChange = (event) => {
        const {name, value} = event.target;
        this.setState({[name]: value});
    }

    handleRegister = (event) => {
        const {email_field: email, password_field: password} = this.state;

        event.preventDefault();
        axiosInstance
            .post('auth/register', {email, password})
            .then(response => {
                notify.show(response.data.message, 'success', 4000);
                this
                    .props
                    .history
                    .push('/login');
                console.log(response);
            })
            .catch(error => {
                if (error.response) {
                    alert(error.response.data.message)
                } else if (error.request) {
                    alert("Request not made")
                }
            });
    }

    render() {
        const {password_field, email_field} = this.state;

        return (
            <div className="background">
                <div id="signbar">
                    <h1>
                        CREATE AN ACCOUNT TO GET STARTED</h1>
                    <form onSubmit={this.handleRegister}>
                        <div className="input-group">
                            <span className="input-group-addon" id="email"><i className='fa fa-envelope'/></span>
                            <input
                                id="emailsignup"
                                name="email_field"
                                aria-describedby="email"
                                className="form-control"
                                placeholder="example@gmail.com"
                                onChange={this.handleInputChange}
                                value={email_field}/>
                        </div>
                        <br/>
                        <div className="input-group">
                            <span className="input-group-addon" id="email"><i className='fa fa-lock'/></span>
                            <input
                                type="password"
                                name="password_field"
                                className="form-control"
                                onChange={this.handleInputChange}
                                value={password_field}
                                placeholder="Password"/>
                        </div>
                        <br/>
                        <div className="row">
                            <div className="col-sm-6 text-left">
                                <Link to="/"><input className="btn-btn-default" value="Cancel" type="button"/></Link>
                            </div>
                            <div className="col-sm-6 text-right">
                                <input name="signup" id="signup" value="Signup" className="btn btn-nav" type="submit"/>
                            </div>
                        </div>

                        <div className="change_link">
                            Already a member ?
                            <Link to="/login">log in
                            </Link>
                        </div>

                    </form>
                </div>
            </div>
        );
    }
}
export default Signup;
