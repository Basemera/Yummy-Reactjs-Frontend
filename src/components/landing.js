import React from 'react';
import { Link } from 'react-router-dom';

const Landing = () => (
  <div className="background-image">
    <div id="header">
      <div id="author"><h3>React Yummy Recipes by KH</h3></div>
      <div id="menu">
        <Link to="/login"><input type="button" id="signin" name="signin" value="LOGIN" /></Link>
        <Link to="/register"><input type="button" id="signup" name="signup" value="SIGNUP " /> </Link>
      </div>
    </div>

    <div id="welcome">
      <h4>You love cooking! then</h4>
      <h3 ><span>WELCOME TO YUMMY RECIPES </span></h3>
      <h4>Go ahead and signup to expore</h4>
    </div>
  </div>
);

export default Landing;
