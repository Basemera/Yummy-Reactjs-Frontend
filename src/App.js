import React, { Component } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import Notifications from 'react-notify-toast';

import Signup from './components/auth/signup';
import Login from './components/auth/login';
import About from './components/auth/About';
import ChangePassword from './components/auth/changePassword';
import Categories from './components/categories/dashboard';
import { Nav } from './components/auth/landing';
import CreateRecipe from './components/recipes/CreateRecipes';

import Landing from './components/auth/landing';
import './css/style.css';

const NotFound = () => (
  <div>Page Not found</div>
);

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div>
          <Nav />
          <Notifications />
          <Switch>
            <Route exact path="/" component={Landing} />
            <Route exact path="/register" component={Signup} />
            <Route exact path="/about" component={About} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/changepassword" component={ChangePassword} />
            <Route exact path="/dashboard" component={Categories} />
            <Route exact path="/recipes" component={CreateRecipe} />
            <Route component={NotFound} />
          </Switch>
        </div>
      </BrowserRouter>

    );
  }
}

export default App;
