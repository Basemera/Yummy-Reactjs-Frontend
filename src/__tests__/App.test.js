import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import App from '../App';
import { Nav } from '../components/auth/landing';
import PrivateRoute from '../components/auth/PrivateRoute';

describe('App component', () => {
  const wrapper = shallow(<App />);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has the the expected components', () => {
    expect(wrapper.find('BrowserRouter')).toHaveLength(1);
    expect(wrapper.find('div').length).toBe(1);
  });

  it('renders a Nav component in the div', () => {
    expect(wrapper.find('div').props().children.find(Nav)).toBeTruthy();
  });

  it('renders the routes', () => {
    expect(wrapper.find('Route').length).toEqual(5);
    expect(wrapper.find('PrivateRoute').length).toEqual(3);
  });

  it('renders the switch', () => {
    expect(wrapper.find('Switch')).toHaveLength(1);
  });

  it('renders the notifications wrapper', () => {
    expect(wrapper.find('_class')).toHaveLength(1);
  });

  it('has the right initial state', () => {
    expect(wrapper.state().loggedin).toBeFalsy();
  });

  it('calls component did mount', () => {
    sinon.spy(App.prototype, 'componentDidMount');
    const fullWrapper = mount(<App />);
    expect(App.prototype.componentDidMount).toHaveProperty('callCount', 1);
    App.prototype.componentDidMount.restore();
  });

  it('fully renders properly', () => {
    const fullWrapper = mount(<App />);
    expect(toJson(fullWrapper)).toMatchSnapshot();
    expect(fullWrapper.find('Landing')).toHaveLength(1);
  });
});
