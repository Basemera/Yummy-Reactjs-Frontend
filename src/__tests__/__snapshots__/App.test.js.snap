// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component fully renders properly 1`] = `
<App>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <Component
          loggedIn={false}
          logout={[Function]}
        >
          <div
            id="header"
          >
            <div
              className="container"
            >
              <div
                className="logo"
              />
              <div
                id="author"
              >
                <h3>
                  Recipes by KH
                </h3>
              </div>
              <div
                id="menu"
              >
                <Link
                  className="btn btn-nav"
                  id="signin"
                  replace={false}
                  to="/login"
                >
                  <a
                    className="btn btn-nav"
                    href="/login"
                    id="signin"
                    onClick={[Function]}
                  >
                    Login
                  </a>
                </Link>
                <Link
                  className="btn btn-nav"
                  id="signup"
                  replace={false}
                  to="/register"
                >
                  <a
                    className="btn btn-nav"
                    href="/register"
                    id="signup"
                    onClick={[Function]}
                  >
                    Signup
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </Component>
        <_class
          options={Object {}}
        >
          <div
            id="notification-wrapper"
          />
        </_class>
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <component
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <Landing
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                loggedIn={false}
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <div
                  className="background-image"
                >
                  <div
                    id="welcome"
                  >
                    <h4>
                      You love cooking! then
                    </h4>
                    <h3>
                      <span>
                        WELCOME TO YUMMY RECIPES
                      </span>
                    </h3>
                    <h4>
                      Go ahead and signup to expore
                    </h4>
                    <div>
                      <h2>
                        It is
                        <Clock>
                          <div
                            className="clock"
                          >
                            12
                            :
                            41
                            :
                            20
                             
                            pm
                          </div>
                        </Clock>
                      </h2>
                    </div>
                  </div>
                </div>
              </Landing>
            </component>
          </Route>
        </Switch>
      </div>
    </Router>
  </BrowserRouter>
</App>
`;

exports[`App component renders properly 1`] = `
<BrowserRouter>
  <div>
    <Component
      loggedIn={false}
      logout={[Function]}
    />
    <_class
      options={Object {}}
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/register"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/reset"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        loggedIn={false}
        path="/about"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        loggedIn={false}
        path="/dashboard"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        loggedIn={false}
        path="/category/:id/recipes"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
